mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/Color.kt

```diff
@@ -1,11 +1,53 @@
 package ch.epfl.sdp.mobile.application.chess
 
-/** The classical colors of a game of chess. */
-enum class Color {
+/**
+ * The classical colors of a game of chess.
+ *
+ * @param normalizeToWhiteDelta a function which normalizes the [Delta].
+ * @param normalizeToWhitePosition a function which normalizes the [Position].
+ * @param opposite a function that returns the opposite color.
+ */
+enum class Color(
+    private val normalizeToWhiteDelta: Delta.() -> Delta,
+    private val normalizeToWhitePosition: Position.() -> Position,
+    private val opposite: () -> Color,
+) {
 
   /** The black color. */
-  Black,
+  Black(
+      normalizeToWhiteDelta = { Delta(x = x, y = -y) },
+      normalizeToWhitePosition = {
+        Position(
+            x = x,
+            y = Board.Size - y - 1, // A simple vertical symmetry.
+        )
+      },
+      opposite = { White },
+  ),
 
   /** The white color. Starts the game. */
-  White,
+  White(
+      normalizeToWhiteDelta = { this },
+      normalizeToWhitePosition = { this },
+      opposite = { Black },
+  );
+
+  /**
+   * Normalizes the given [Delta].
+   *
+   * @param delta the [Delta] to normalize.
+   * @return the normalized delta.
+   */
+  fun normalize(delta: Delta): Delta = normalizeToWhiteDelta(delta)
+
+  /**
+   * Normalizes the given [Position].
+   *
+   * @param position the [Position] to normalize.
+   * @return the normalized position.
+   */
+  fun normalize(position: Position): Position = normalizeToWhitePosition(position)
+
+  /** Returns the [Color] of the adversary. */
+  fun other(): Color = opposite()
 }
```

Comment:
Since both Black and White get normalized, maybe the method `normalizeToWhitePosition` should actually be called `normalizePosition`, same thing with `normalizeDelta`?

Reply:
Fixed in affda0427dcb613a4afb6ef6c53c787c5213f072.